cairo/tests/utils/args_gen.py

```diff
@@ -1400,12 +1400,15 @@ def generate_trie_arg(
     # In case of a Trie, we need the trie.default to be the default value of the dict.
     dict_ptr = segments.memory.get(data)
 
-    if isinstance(dict_manager, DictManager):
-        default_value = dict_manager.trackers[
-            dict_ptr.segment_index
-        ].data.default_factory()
+    if isinstance(arg._data, defaultdict):
+        if isinstance(dict_manager, DictManager):
+            default_value = dict_manager.trackers[
+                dict_ptr.segment_index
+            ].data.default_factory()
+        else:
+            default_value = dict_manager.get_default_value(dict_ptr.segment_index)
```

Comment:
i don't understand when dict_manager isn't of class `DictManager`

Reply:
when it's a `RustDictManager`